optuna/storages/_rdb/alembic/versions/v3.0.0.d.py

```diff
@@ -0,0 +1,167 @@
+    # does not support such a SQL statement. So first add a column with schema-level
+    # default value setting, then remove it by `batch_op.alter_column()`.
+    with op.batch_alter_table("trial_values") as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "value_type",
+                sa.Enum("FINITE", "INF_POS", "INF_NEG", name="trialvaluetype"),
+                nullable=False,
+                server_default="FINITE",
+            ),
+        )
+    with op.batch_alter_table("trial_values") as batch_op:
+        batch_op.alter_column("value_type", server_default=None)
+        batch_op.alter_column(
+            "value",
+            existing_type=sa.Float(precision=FLOAT_PRECISION),
+            existing_nullable=False,
+            nullable=True,
+        )
+
+    session = orm.Session(bind=bind)
+    try:
+        records = session.query(TrialValueModel).all()
+        mapping = []
+        for r in records:
+            value: float
+            if np.isclose(r.value, RDB_MAX_FLOAT) or np.isposinf(r.value):
+                value = float("inf")
+            elif np.isclose(r.value, RDB_MIN_FLOAT) or np.isneginf(r.value):
+                value = float("-inf")
+            elif np.isnan(r.value):
+                value = float("nan")
+            else:
+                value = r.value
+            (
+                stored_value,
+                float_type,
+            ) = TrialValueModel.value_to_stored_repr(value)
+            mapping.append(
+                {
+                    "trial_value_id": r.trial_value_id,
+                    "value_type": float_type,
+                    "value": stored_value,
+                }
+            )
+        session.bulk_update_mappings(TrialValueModel, mapping)
+        session.commit()
+    except SQLAlchemyError as e:
+        session.rollback()
+        raise e
+    finally:
+        session.close()
+
+
+def downgrade():
+    bind = op.get_bind()
+    session = orm.Session(bind=bind)
+
+    try:
+        records = session.query(TrialValueModel).all()
+        mapping = []
+        for r in records:
+            if r.value_type == TrialValueModel.TrialValueType.FINITE:
+                continue
+
+            _value = r.value
+            if r.value_type == TrialValueModel.TrialValueType.INF_POS:
+                _value = RDB_MAX_FLOAT
+            else:
+                _value = RDB_MIN_FLOAT
+
+            mapping.append(
+                {
+                    "trial_value_id": r.trial_value_id,
+                    "value": _value,
+                }
+            )
+        session.bulk_update_mappings(TrialValueModel, mapping)
+        session.commit()
+    except SQLAlchemyError as e:
+        session.rollback()
+        raise e
+    finally:
+        session.close()
+
+    with op.batch_alter_table("trial_values", schema=None) as batch_op:
+        batch_op.drop_column("value_type")
```

Comment:
I think we should make `value` column `nullable=False` here.

Reply:
Sorry, we didn't share the discussion with you. When `value_type` is not `FINITE`, we put `NULL` in `value` column. In discussion among me, @toshihikoyanase and @HideakiImamura, we agreed that that is more easy to understand than setting `value` as non-nullable and putting e.g. 0.0.

Reply:
I may miss the point, but `value` does not accept `null` before we migrate the schema, so we should revert `nullable` of `value` in this `downgrade` function.

Reply:
Sorry, I was misunderstood. Thanks for the clarification!